common/interpolation/include/interpolation/spline_interpolation.hpp

```diff
@@ -49,6 +49,9 @@ struct MultiSplineCoef
 std::vector<double> slerp(
   const std::vector<double> & base_keys, const std::vector<double> & base_values,
   const std::vector<double> & query_keys);
+std::array<std::vector<double>, 3> slerp(
+  const std::vector<double> & base_keys, const std::vector<double> & base_x_values,
+  const std::vector<double> & base_y_values, const std::vector<double> & query_keys);
```

Comment:
spline_interpolation.hpp is pure spline interpolation implementation.
If you create this function like for point2d, could you move it to spline_interpolation_points_2d.cpp and rename it easier name to understand like slerp2dFromXY (I don't know if this name is easy to understand though).